pvlib/location.py

```diff
@@ -72,11 +72,12 @@ def __init__(self, latitude, longitude, tz='UTC', altitude=None,
             self.tz = 'UTC'
             self.pytz = pytz.UTC
         elif isinstance(tz, datetime.tzinfo):
+            # This includes pytz timezones.
             self.tz = tz.zone
-            self.pytz = tz
+            self.pytz = pytz.timezone(tz.zone)
         elif isinstance(tz, (int, float)):
-            self.tz = tz
```

Comment:
There was a reason for storing the number here, but it's probably traceable all the way back to Matlab and no longer relevant.

Reply:
I am open to only storing the `pytz`, but we might provide a read-only `tz` property that prints the IANA `str`. (See my other response, where I say that I also don't like the duplication of state.)